ui/components/MesheryMeshInterface/PatternService/index.js

```diff
@@ -64,6 +64,22 @@ function PatternService({ formData, jsonSchema, onChange, type, onSubmit, onDele
       });
     return sortedProperties;
   };
+  // added a p_type="array" to every object in an array to override the normal behaviour of objects.
+  const addPTypeToArray = (properties) => {
+    Object.keys(properties).forEach(key => {
+      if (properties[key]?.items?.type === "object") {
+        properties[key].items.p_type = "array";
+      }
+      if (properties[key]?.properties) {
+        properties[key].properties = addPTypeToArray(properties[key].properties);
+      }
+      if (properties[key]?.items?.properties) {
+        properties[key].items.properties = addPTypeToArray(properties[key].items.properties);
+      }
+    });
+    return properties;
+  };
+  addPTypeToArray(jsonSchema.properties);
```

Comment:
This runs everytime there is any re-render, you can check it by adding a console or debuggin, but this would be less performant. It needs to be done only "once" per schema. Secondly, there is a "schemaBuilder" where such checking/modification in rjsf schema is being done, can it be done there? What do you think?

Reply:
@Abhishek-kumar09 should we do this in Cuelang or Golang? // @gr455 are you familiar with this area, yet?

Reply:
@Abhishek-kumar09 The above logic run once per render, until you close the form, and open it back. If we move the logic to schema builder, the behaviour remains same run once per render.

Reply:
Okay,then its fine. Let it stay here.
The function definition isn't dependent on any internal react state, so can you move that out of the react component?

Reply:
> @Abhishek-kumar09 should we do this in Cuelang or Golang? // @gr455 are you familiar with this area, yet?

Yes it would make more sense there.

Reply:
Ok